routes/index.js

```diff
@@ -14,12 +16,36 @@ router.get("/login", function (req, res, next) {
   res.render("login");
 });
 
-router.get('/signup/phone', function (req, res, next) {
-  res.render('signup-phone', { title: '배민 - 전화번호 인증'})
+router.get("/signup/phone", function (req, res, next) {
+  res.render("signup-phone", { title: "배민 - 전화번호 인증" });
 });
 
 router.get("/signup/detail", function (req, res, next) {
-  res.render("signup-detail");
+  const hasSignupError = req.session?.signupError;
+  if (!hasSignupError) {
+    res.render("signup-detail", { error: null });
+    return;
+  }
+
+  req.session.signupError = null;
+  res.render("signup-detail", { error: hasSignupError });
```

Comment:
```suggestion
  const signupError = req.session?.signupError;
  if (signupError) {
    delete req.session.signupError;
    res.render("signup-detail", { error: signupError });
    return;
  }
  
  res.render("signup-detail");
```

session에 signupError를 저장했고 이를 꺼내서 template에 던지는 것이기에 `hasSignupError`가 아닌 signupError로 꺼내는 것이 맞다고 생각합니다.

또한 signupError가 존재할 경우에만 session에서 signupError를 제거하고 보내고 없을 경우에는 일반적으로 렌더하는게 좋겠습니다.

Reply:
멋진 의견 감사합니다~~ :)